docs/SIL.rst

```diff
@@ -1835,6 +1835,139 @@ This is a form of ownership that is used to model two different use cases:
   trivial pointer to a class. In that case, since we have no reason to assume
   that the object will remain alive, we need to make a copy of the value.
 
+Forwarding Instructions
+~~~~~~~~~~~~~~~~~~~~~~~
+
+NOTE: In the following, we assumed that one read the section above, `Value Ownership Kind`_.
+
+In OSSA, certain SIL instructions derive the value ownership kinds of their
+results from the results type and the value ownership kind of a subset of said
+instruction's operands. This computation of the value ownership kind of the
+result from the set of operands is called "forwarding" and is performed by
+inferring the result's ownership when the instruction is constructed. As a
+result, we call these instructions "forwarding instructions". This results in
+two properties:
+
+* When manipulating forwarding instructions programatically, one must manually
+  update their forwarded ownership since most of the time the ownership is
+  stored in the instruction itself. Don't worry though because the SIL verifier
+  will catch this error for you if you forget to do so!
+
+* Textual SIL does not represent the ownership of forwarding instructions
+  explicitly. Instead, the instruction's ownership is inferred normally from the
+  parsed operand. Since the SILVerifier runs on Textual SIL after parsing, you
+  can feel confident that ownership constraints were inferred correctly.
+
+Forwarding has slightly different ownership semantics depending on the value
+ownership kind of the operand on construction and the result's type. We go
+through each below:
+
+* Given an ``@owned`` operand, the forwarding instruction is assumed to end the
+  lifetime of the operand and produce an ``@owned`` value if non-trivially typed
+  and ``@any`` if trivially typed. Example: This is used to represent the
```

Comment:
This doesn't make any sense to me. Trivial types have no ownership, not `@any`, whatever that is. I'll stop here, because I can't parse this description.

Reply:
Let me be more clear.

There are three basic kinds of ownership: `none`, `+1`, `+0`.
There are three ownership constraints: `@trivial`, `@owned`, `@guaranteed`
We can relate ownership kinds and ownership constraints by representing them in the same lattice.
We can make the lattice symmetric by giving it an 'unknown' state. But values never have an unknown ownership kind--trivial values are actually the *inverse* of unknown.

We can say that an operation with no constraints actually has an `@any` constraint, which *does* correspond to 'unknown'. So this description is backwards because `@any` really corresponds to the inverse of a trivial type.

Applying a constraint simply performs a meet of the kind and the constraint. If the meet returns the same state as the constraint, then the constraint is satisfied.

Reply:
I meant to say @none instead of @any. I just mistyped.